### PR TITLE
GEOMESA-946 AND queries against indexed attributes do not return correct results

### PR DESCRIPTION
**Pull request by Afterglow375**
Signed-off-by: Alex Tatusko <atatusko@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala [hunk 1]
```diff
@@ -590,6 +590,52 @@ class AttributeIndexStrategyTest extends Specification with TestWithDataStore {
...
     }
+
+    "correctly query on indexed attributes with nonsensical AND queries" >> {
+      "redundant int query" >> {
+        val features = execute("age > 25 AND age > 15")
+        features must haveLength(1)
+        features must contain("bob")
+      }
+
+      "int query that returns nothing" >> {
```
Comment (cne1x) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala]: What does "int" mean in this context?

**Comment by elahrvivaz** [comment]
heh, so are there still some cases that fail? if not, maybe change the commit message to be something along the lines of 'fixing AND queries against indexed attributes'

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala (hunk 1 above)
Comment (Afterglow375) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala]: int as in "age" is an Integer in the SFT.

Comment (cne1x) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala]: Sigh.  I could have sworn I saw that label on a test that was doing date filtering.  The longer I stay this afternoon, the more I realize I'm still fogged.  Sorry for the silly question.

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/AttributeIndexStrategyTest.scala]: @cne1x check the commit history ;)

**Comment by cne1x** [comment]
Merged as commit b81c9f810aaae7b720f7a77af9d1fdd40529e166

**Pull request closed without merging**